tests/ut/ut_scargo_gen.py

```diff
@@ -0,0 +1,307 @@
+            certs_mode=None,
+            certs_input=None,
+            certs_passwd=None,
+            fs=False,
+            single_bin=False,
+        )
+
+        # test if expected files was generated
+        assert os.path.exists(
+            f"tests/mocks/{NESTED_LIB_FILE_LOCATION}/{NAME_OF_LIB_HPP_FILE}"
+        )
+        assert os.path.exists(
+            f"tests/mocks/{NESTED_LIB_FILE_LOCATION}/mock_{NAME_OF_LIB_HPP_FILE}"
+        )
+        assert os.path.exists(
+            f"tests/mocks/{NESTED_LIB_FILE_LOCATION}/mock_{NAME_OF_LIB_HPP_FILE}".replace(
+                ".hpp", ".cpp"
+            )
+        )
+
+
+@pytest.mark.parametrize(
+    "file",
+    ["test_lib.cpp", NAME_OF_TEST_TXT_FILE, NAME_OF_TEST_NO_EXTENSION_FILE],
+    ids=["cpp file", "txt file", "file_with_no_extension"],
+)
+def test_gen_mock_for_unexpected_extension_file(
+    coppy_test_project_stm32: Path,
+    mocker: MockerFixture,
+    file: str,
+) -> None:
+    """
+    This test check if for .h located in src dictionary mock files will be created under tests/mocks dictionary
+    as a result of scargo gen command.
+    """
+    os.chdir(coppy_test_project_stm32)
+    project_add_extra_test_files(coppy_test_project_stm32 / "src")
+    path_to_file = Path(os.getcwd(), "src", file)
+    # make sure that files which needs to be generated do not exist - precondition
+    assert not os.path.exists(f"tests/mocks/{file}")
+
+    mocker.patch(
+        f"{scargo_gen.__module__}.prepare_config",
+        return_value=get_scargo_config_or_exit(),
+    )
+    with pytest.raises(SystemExit) as scargo_gen_sys_exit:
+        scargo_gen(
+            profile="Debug",
+            gen_ut=None,
+            gen_mock=path_to_file,
+            certs=None,
+            certs_mode=None,
+            certs_input=None,
+            certs_passwd=None,
+            fs=False,
+            single_bin=False,
+        )
+    assert scargo_gen_sys_exit.type == SystemExit
+    assert scargo_gen_sys_exit.value.code == 1
+
+    # test if expected files was generated
+    assert not os.path.exists(f"tests/mocks/{file}")
+
+
+@pytest.mark.skip("Not implemented yet")
+class TestGenCerts:
+    """This class collects all unit tests for scargo_gen command with certs option"""
+
+    def test_en_certs_simple(
+        self,
+        request: pytest.FixtureRequest,
+        coppy_test_project_stm32: Path,
+        mocker: MockerFixture,
+        file: str,
+    ) -> None:
+        """This test simply check if command gen certs create certs files"""
+        pass
+
+
+@pytest.mark.parametrize("target", ["x86", "stm32", "esp32"])
+def test_gen_ut_new_project(tmp_path: Path, target: str, mocker: MockerFixture) -> None:
+    test_data = Path(__file__).parents[1].joinpath("test_data")
+    os.environ["IDF_PATH"] = Path(test_data, "esp32_spiff").as_posix()
+    target_project = Target.get_target_by_id(target)
+    os.chdir(tmp_path)
+    project_name = "test_project_gen_ut"
+    scargo_new(
```

Comment:
If it is possible it will be nice to use fixtures from conftest like: `new_project_x86` instead of crating new projects each time.